### PR TITLE
NEW: Add technical message to error page on dev environments

### LINKED ISSUE #30: Exception message is discarded when using httpError
If a developer calls `return this->httpError(401, 'Some message for unauthorised users');`, a HTTPResponse_Exception is thrown. If an ErrorPage is not loaded, the user will receive a barebones controller response with a 401 that says "Some message for unauthorised users", which is obtained from the `Exception::getMessage()` method. 

This message seems to get discarded somewhere when an ErrorPage is generated. It is not available to a published ErrorPage, and thus cannot be displayed on a template if your theme implements one.

### PR DESCRIPTION
**Pull request by sminnee**
The formatting its pretty simple - just a bolded paragraph. It’s kept
off live environments to avoid inappropriate information leakage.

Fixes #30.

### CODE REVIEW

**Comment by elliot-sawyer** [comment]
Mentioned on the other issue:

Sometimes there are multiple reasons why someone might get a particular HTTP code, and the message might be user friendly. A developer might want to display the message by implementing it in the template, which can't be done if the site is forced into dev mode. 

`$errorPage->ErrorMessage = $errorMessage;` would be fairly harmless to add. If a dev wants to risk the info leak, it's up to them to do so.

**Comment by sminnee** [comment]
I've incorporated your suggestion @elliot-sawyer

**Comment by elliot-sawyer** [comment]
Awesome!  Looks good!

File: src/ErrorPage.php [hunk 1]
```diff
@@ -102,6 +105,17 @@ public static function response_for($statusCode)
             Requirements::clear();
             Requirements::clear_combined_files();
 
+            if ($errorMessage) {
```
Comment (Firesphere) [inline src/ErrorPage.php]: I would like to be able to opt out of the message in `dev` mode. Not always does it add value and sometimes it hides the actual information?

Comment (elliot-sawyer) [reply src/ErrorPage.php]: I've raised a PR on @sminnee 's fork to address this

Comment (elliot-sawyer) [reply src/ErrorPage.php]: @firesphere I've added the ability to opt out to @sminnee 's PR.  Are you happy with the proposed changes?

Comment (elliot-sawyer) [reply src/ErrorPage.php]: ...after I fix the broken tests, of course!

**Pull request merged**